core/src/main/java/org/elasticsearch/script/TemplateService.java

```diff
@@ -0,0 +1,185 @@
+                return scriptMetadata.getCode();
+            }
+
+            @Override
+            protected StoredScriptSource lookupStoredScript(ClusterState clusterState,
+                    String cacheKey) {
+                if (clusterState == null) {
+                    return null;
+                }
+                MetaData metaData = clusterState.metaData();
+                if (metaData == null) {
+                    return null;
+                }
+                ScriptMetaData scriptMetaData = clusterState.metaData().custom(ScriptMetaData.TYPE);
+                if (scriptMetaData == null) {
+                    return null;
+                }
+
+                String id = cacheKey;
+                // search template requests can possibly pass in the entire path instead
+                // of just an id for looking up a stored script, so we parse the path and
+                // check for appropriate errors
+                String[] path = id.split("/");
+
+                if (path.length == 3) {
+                    id = path[2];
+
+                    deprecationLogger.deprecated("use of </lang/id> [{}] for looking up stored "
+                            + "scripts/templates has been deprecated, use only <id> [{}] instead",
+                            cacheKey, id);
+                } else if (path.length != 1) {
+                    throw new IllegalArgumentException( "illegal stored script format [" + id
+                            + "] use only <id>");
+                }
+
+                return scriptMetaData.getStoredScript(id, backend.getExtension());
+            }
+
+            @Override
+            protected boolean areAnyScriptContextsEnabled(String cacheKey, ScriptType scriptType) {
+                for (ScriptContext scriptContext : scriptContextRegistry.scriptContexts()) {
+                    if (scriptPermits.checkContextPermissions(backend.getType(), scriptType,
+                            scriptContext)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            protected void checkContextPermissions(String cacheKey, ScriptType scriptType,
+                    ScriptContext scriptContext) {
+                if (scriptPermits.checkContextPermissions(backend.getType(), scriptType,
+                        scriptContext) == false) {
+                    throw new IllegalStateException("templates of [" + scriptType + "],"
+                            + " operation [" + scriptContext.getKey() + "] are disabled");
+                }
+            }
+
+            @Override
+            protected void checkCompilationLimit() {
+                scriptPermits.checkCompilationLimit();
+            }
+
+            @Override
+            protected CompiledScript compile(ScriptType scriptType, String cacheKey) {
+                Object compiled = backend.compile(null, cacheKey, emptyMap());
```

Comment:
I believe this line is missing the crucial "Content-Type" so I'll need to rework some things on Monday....